src/sidebar/helpers/annotation-user.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * @typedef {import("../../types/api").Annotation} Annotation
+ */
+
+import { username } from './account-id';
+
+/**
+ * What string should we use to represent the author (user) of a given
+ * annotation: a display name or a username?
+ *
+ * The nice, human-readable display name should be used when a display_name
+ * is available on the annotation AND:
+ * - The author (user) associated with the annotation is a third-party user, OR
+ * - The `client_display_names` feature flag is enabled
+ *
+ * Return the string that should be used for display on an annotation: either the
+ * username or the display name.
+ *
+ * @param {Pick<Annotation, 'user'|'user_info'>} annotation
+ * @param {boolean} isThirdPartyUser - Is the annotation's user third-party?
```

Comment:
There is a small hazard with multiple boolean args that it is easy to mix up the argument order. I think its OK for the moment as there are only two of them.

Reply:
I agree. I agonized lightly over this one. Open to any suggestions that might leap into your head.